### ISSUE TITLE
Support PowerShell as the default shell on Unix-like platforms

### ISSUE THREAD

**Issue by mklement0** [Microsoft/vscode#60971]
<sup>Migrated and generalized from https://github.com/OmniSharp/omnisharp-vscode/issues/2605</sup>

It would be great if Visual Studio Code also supported a Unix configuration where PowerShell (Core) is configured as the user's default shell.

Below is an example of where that currently fails: the debugger-invocation command line assumes a POSIX-like shell on Unix, and submits a command using the latter's syntax, which breaks in PowerShell.

There may be other, related cases.

In the debugger-invocation case, only a trivial tweak is required to support PowerShell too (see below).

## Steps to reproduce

* On a Unix-like platform with PowerShell Core installed, make the latter your default shell as follows: `chsh -s /usr/local/bin/pwsh` (macOS) or `chsh -s /usr/bin/pwsh` (Linux).

* On Linux, reboot to make the new default shell take effect (not necessary on macOS).

* Open any C# project in VS Code and debug with a `".NET Core Launch (console)"` configuration with setting `"console": "externalTerminal"`

## Desired  behavior

Debugger invocation should succeed.

## Current behavior

Debugger invocation fails, because the command line sent to the terminal assumes POSIX-like-shell syntax (e.g., `bash`), which PowerShell doesn't support.

Sample symptom:

```powershell
PS /Users/jdoe> cd '/Users/jdoe/.vscode/extensions/ms-vscode.csharp-1.16.2/.debugger/';  '/Users/jdoe/.vscode/extensions/ms-vscode.csharp-1.16.2/.debugger/vsdbg' '--interpreter=vscode' '--connection=/var/folders/19/0lxcl7hd63d6fqd813glqppc0000gn/T/CoreFxPipe_vsdbg-ui-9bb838baf07245f591dda2adcfd2536f'
At line:1 char:155
+ ... -vscode.csharp-1.16.2/.debugger/vsdbg' '--interpreter=vscode' '--conn ...
+                                            ~~~~~~~~~~~~~~~~~~~~~~
Unexpected token ''--interpreter=vscode'' in expression or statement.
At line:1 char:178
+ ... ter=vscode' '--connection=/var/folders/19/0lxcl7hd63d6fqd813glqppc000 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Unexpected token ''--connection=/var/folders/19/0lxcl7hd63d6fqd813glqppc0000gn/T/CoreFxPipe_vsdbg-ui-9bb838baf07245f591dda2adcfd2536f'' in expression or statement.
+ CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : UnexpectedToken
```

With a _quoted_ path to `vsdbg`, the call would have to be prefixed with `& `  to make PowerShell happy.

**A pragmatic workaround would be _not_ to quote the `vsdbg` path, which would make the call succeed in PowerShell too, without special-casing.**

This should be safe as long as the user's `$HOME` path has no embedded spaces - something I've never seen in the Unix world.

**Comment by weinand** [comment]
Clarification:
PowerShell on Unix-like platforms is already supported for VS Code's integrated terminal:

![2018-11-07_10-15-01](https://user-images.githubusercontent.com/1898161/48121732-043e2500-e276-11e8-9b83-34a866d581ce.png)

I'm using these settings to configure the integrated terminal to use PowerShell:
```
    "terminal.integrated.shell.osx": "/usr/local/bin/powershell",
    "terminal.integrated.shellArgs.osx": []
```

Without this setting VS Code doesn't know how to preprocess command line arguments correctly.

What you are asking for is to support the same for the "external terminal", e.g. the launch config setting `"console": "externalTerminal"`.

**Comment by mklement0** [comment]
Yes, this issue is about the _external_ terminal (as noted in the "Steps to reproduce" section, but I've updated the description at the top to make that clearer).

Two asides:

* The Powershell Core binary was renamed to `pwsh` before the first public release (`/usr/local/bin/pwsh` on macOS).

* The need to override the default `"terminal.integrated.shellArgs.osx"` value (suppress the `-l`) will hopefully go away soon: https://github.com/PowerShell/PowerShell/issues/7704
